### PR TITLE
feat: the setting ui for new dashboard

### PR DESCRIPTION
**Pull request by yanzhihong23**
image

### CODE REVIEW

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 1]
```diff
@@ -136,6 +136,7 @@ const ShapeContainer = styled('div')(({ theme }) => ({
 export interface PageFrameProps extends React.PropsWithChildren<{}> {
     title: React.ReactNode | string
     primaryAction?: React.ReactNode
+    withoutShapeContainer?: boolean
```
Comment (Jack-Works) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:139]: Please contact the designer. If the inconsistency here is by design or a mistake.

File: packages/dashboard/src/pages/Settings/components/SettingSwitch.tsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { Switch, experimentalStyled as styled } from '@material-ui/core'
+
+export default styled(Switch)(() => ({}))
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/components/SettingSwitch.tsx:3]: 👀 So why not remove this

File: packages/dashboard/src/pages/Settings/components/AncientPostsSetting.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import type { ChangeEvent } from 'react'
+import { Services } from '../../../API'
+import { useAncientPostsCompatibilityMode } from '../api'
+import SettingSwitch from './SettingSwitch'
+
+export default function AncientPostsSetting() {
+    let checked = useAncientPostsCompatibilityMode()
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/components/AncientPostsSetting.tsx:7]:
```suggestion
    const checked = useAncientPostsCompatibilityMode()
```

File: packages/dashboard/src/pages/Settings/components/AppearanceSetting.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { MenuItem } from '@material-ui/core'
+import { Services } from '../../../API'
+import { useAppearance } from '../../Personas/api'
+
+import SettingSelect from './SettingSelect'
+
+export default function AppearanceSetting() {
+    let mode = useAppearance()
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/components/AppearanceSetting.tsx:8]:
```suggestion
    const mode = useAppearance()
```

File: packages/dashboard/src/pages/Settings/components/AppearanceSetting.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    const handleChange = (event: any) => {
+        Services.Settings.setTheme(event.target.value)
+    }
+
+    return (
+        <SettingSelect value={mode} onChange={handleChange}>
+            <MenuItem value="default">Follow system settings</MenuItem>
+            <MenuItem value="light">Light</MenuItem>
+            <MenuItem value="dark">Dark</MenuItem>
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/components/AppearanceSetting.tsx:17]: Although it isn't likely to being changed, but I think it's better to use 
```
<MenuItem value={Appearance.default}>Light</MenuItem>
```

here.

`Appearance` is defined in the maskbook package. I think we can move it to the `theme` package.

File: packages/dashboard/src/pages/Settings/components/DataSourceSetting.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    const handleChange = (event: any) => {
+        const value = event.target.value
+        Services.Settings.setTrendingDataSource(value)
+    }
+
+    return (
+        <SettingSelect value={source} onChange={handleChange}>
+            <MenuItem value={0}>CoinGecko</MenuItem>
+            <MenuItem value={1}>CoinMarketCap</MenuItem>
+            <MenuItem value={2}>Uniswap Info</MenuItem>
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/components/DataSourceSetting.tsx:17]: The same as above, use the enum from `DataProvider`. (It is not defined in `shared` package. So you can leave a TODO here, once #3215 has merged, DataProvider enum should be moved in that package)

File: packages/dashboard/src/pages/Settings/components/LanguageSetting.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { MenuItem } from '@material-ui/core'
+import { useLanguage } from '../api'
+import { Services } from '../../../API'
+import SettingSelect from './SettingSelect'
+
+export default function LanguageSetting() {
+    let lang = useLanguage()
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/components/LanguageSetting.tsx:7]:
```suggestion
    const lang = useLanguage()
```

File: packages/dashboard/src/pages/Settings/components/LanguageSetting.tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    const handleChange = (event: any) => {
+        Services.Settings.setLanguage(event.target.value)
+    }
+
+    return (
+        <SettingSelect value={lang} onChange={handleChange}>
+            <MenuItem value="zh">中文</MenuItem>
+            <MenuItem value="en">English</MenuItem>
+            <MenuItem value="ko">한국인</MenuItem>
+            <MenuItem value="ja">日本語</MenuItem>
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/components/LanguageSetting.tsx:17]: The same as above. And enum `Language` should be moved to the `theme` package or `shared` IMO.

File: packages/dashboard/src/pages/Settings/index.tsx [hunk 9]
```diff
@@ -1,10 +1,105 @@
...
+import PasswordSetting from './components/PasswordSetting'
+import LogoutSetting from './components/LogoutSetting'
+import DataSourceSetting from './components/DataSourceSetting'
 
 export default function Settings() {
+    const t = useDashboardI18N()
     return (
-        <PageFrame title="Settings" primaryAction={<Button>Create a new wallet</Button>}>
-            <Typography>Hi</Typography>
+        <PageFrame title={t.settings()} withoutShapeContainer={true}>
```
Comment (Jack-Works) [inline packages/dashboard/src/pages/Settings/index.tsx:34]:
```suggestion
        <PageFrame title={t.settings()} withoutShapeContainer>
```

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/components/DashboardFrame/index.tsx (hunk 1 above)
Comment (yanzhihong23) [reply packages/dashboard/src/components/DashboardFrame/index.tsx:139]: This is by design

File: packages/dashboard/src/pages/Settings/components/SettingSwitch.tsx (hunk 2 above)
Comment (yanzhihong23) [reply packages/dashboard/src/pages/Settings/components/SettingSwitch.tsx]: style updated

File: packages/dashboard/src/pages/Settings/components/AncientPostsSetting.tsx (hunk 3 above)
Comment (yanzhihong23) [reply packages/dashboard/src/pages/Settings/components/AncientPostsSetting.tsx]: done

File: packages/dashboard/src/pages/Settings/components/AppearanceSetting.tsx (hunk 5 above)
Comment (yanzhihong23) [reply packages/dashboard/src/pages/Settings/components/AppearanceSetting.tsx]: done

File: packages/dashboard/src/components/DashboardFrame/index.tsx [hunk 10]
```diff
@@ -173,7 +177,8 @@ export const PageFrame = memo((props: PageFrameProps) => {
                     </NavigationDrawer>
                 )}
                 <ShapeHelper>
-                    <ShapeContainer>
+                    <ShapeContainer
+                        className={props.noBackgroundFill ? undefined : classes.shapeContainerWithBackground}>
```
Comment (nuanyang233) [inline packages/dashboard/src/components/DashboardFrame/index.tsx:181]: You can use [classnames](https://www.npmjs.com/package/classnames)

Comment (Jack-Works) [reply packages/dashboard/src/components/DashboardFrame/index.tsx:181]: I think this is ok for now

File: packages/dashboard/src/pages/Settings/components/SettingSelect.tsx [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import { MaskColorVar } from '@dimensiondev/maskbook-theme'
+import { Select, experimentalStyled as styled, selectClasses, outlinedInputClasses } from '@material-ui/core'
+
+export default styled(Select)(() => ({
+    [`& .${selectClasses.select}.${selectClasses.outlined}`]: {
+        padding: '12.5px 70px 12.5px 14px !important',
```
Comment (nuanyang233) [inline packages/dashboard/src/pages/Settings/components/SettingSelect.tsx:6]: please use the `theme.spacing`

Comment (Jack-Works) [reply packages/dashboard/src/pages/Settings/components/SettingSelect.tsx:6]: Actually, I have a doubt about this now. 🤔 If we're not using the dynamic spacing and the design is not following the dynamic spacing system, why we should stick on those...

Comment (nuanyang233) [reply packages/dashboard/src/pages/Settings/components/SettingSelect.tsx:6]: Ok, if we're not using dynamic spacing, We really don't need to stick on these

**Review by nuanyang233 — APPROVED** [review]

**Pull request merged**